pixel-saver@deadalnix.me/extension.js

```diff
@@ -44,20 +44,36 @@
  * [5]:http://www.webupd8.org/2011/05/how-to-remove-maximized-windows.html
  *
  */
+const Lang = imports.lang;
 const ExtensionUtils = imports.misc.extensionUtils;
 const Me = ExtensionUtils.getCurrentExtension();
 const Decoration = Me.imports.decoration;
 const Buttons = Me.imports.buttons;
 const AppMenu = Me.imports.app_menu;
+const Settings = Me.imports.settings;
+
+let options;
 
 function init(extensionMeta) {
 	Buttons.init(extensionMeta);
 	Decoration.init(extensionMeta);
 	AppMenu.init(extensionMeta);
 }
 
+function loadOptions() {
+	options = new Settings.Options();
+	options.BUTTON_POSITION.changed(Lang.bind(this, function() {
+		if (options.BUTTON_POSITION.get() == Buttons.Position.HIDDEN) {
+			Buttons.hideButtons();
+		} else {
+			Buttons.showButtons();
+		}
```

Comment:
It's no duplication. At one point we have to react to settings changes. Where do you see duplication here?

Loading the settings here seems preferable than moving it into Buttons.js because there might be more preferences in the future (I have already more in my private fork) and this way you have once central place where the wiring is done. Extension.js kind of acts as the central controller.